core/src/mindustry/world/blocks/power/Battery.java

```diff
@@ -1,10 +1,39 @@
 package mindustry.world.blocks.power;
 
+import arc.*;
+import arc.graphics.Color;
+import arc.graphics.g2d.*;
+import mindustry.entities.type.TileEntity;
+import mindustry.world.*;
+import mindustry.world.modules.PowerModule;
+
 public class Battery extends PowerDistributor{
+    public TextureRegion lightsRegion;
+    public Color lightColor = Color.valueOf("fb9567");
 
     public Battery(String name){
         super(name);
         outputsPower = true;
         consumesPower = true;
     }
+    
+    @Override
+    public void load(){
+        super.load();
+        lightsRegion = Core.atlas.find(name + "-lights");
+    }
+
+    @Override
+    public void draw(Tile tile){
+        super.draw(tile);
+        TileEntity ent = tile.ent();
+        PowerModule entity = ent.power;
```

Comment:
Don't create a variable if you're only going to use it once; use this directly instead.

Reply:
ok, will do
be doesn't seem to compile